src/topology/algebra/group.lean

```diff
@@ -746,3 +746,100 @@ instance : topological_group (units α) :=
     (@continuous_embed_product α _ _))).prod_mk (continuous_op.comp continuous_coe)) }
 
 end units
+
+/-!
+### Lattice of group topologies
+We define a type class `group_topology α` which endows a group `α` with a topology such that all
+group operations are continuous.
+
+Group topologies on a fixed group `α` are ordered, by reverse inclusion. They form a complete
+lattice, with `⊥` the discrete topology and `⊤` the indiscrete topology.
+
+Any function `f : α → β` induces `coinduced f : topological_space α → group_topology β`. -/
+
+/-- A group topology on a group `α` is a topology for which multiplication and inversion
+are continuous. -/
+@[ext]
+structure group_topology (α : Type u) [group α]
```

Comment:
Could you please provide the additive variant as well? Almost all of it should be generatable (is that a word?) by the `to_additive` tactic.

Reply:
I tried using `to_additive` in the definition of `group_topology` and the lemmas after it, but I kept getting the error `@[to_additive] failed. Type mismatch in additive declaration.
For help, see the docstring of to_additive.attr, section `Troubleshooting`.
Failed to add declaration ...`.
I don't think `to_additive` is doing what I want in the definition; `#print add_group_topology` gives me `@[_ext_core id.{1} name group_topology.ext]
constant add_group_topology : Π (α : Type u) [_inst_1 : add_group α], Type u`, but I was expecting something like 
```
Π (α : Type u) [_inst_1 : add_group α], Type u
fields:
add_group_topology.to_topological_space : Π {α : Type u} [_inst_1 : add_group α], add_group_topology α → topological_space α
add_group_topology.to_topological_add_group : ∀ {α : Type u} [_inst_1 : add_group α] (self : add_group_topology α), topological_add_group α
```
That's why I only did the PR for `group` - but I can adapt it to `add_group` by hand.

Reply:
It's possible I'm just not using `to_additive` correctly in the definition; this is what I had:
```
@[ext, to_additive]
structure group_topology (α : Type u) [group α]
  extends topological_space α, topological_group α : Type u
```

Reply:
Unfortunately `to_additive` doesn't work on structures, so you need to give the definition of `add_group_topology` yourself first before `group_topology` with `to_additive` on group topology still.
After that it should work fine on all the following lemmas though, without the need to give duplicate versions of them all.

Reply:
Ah, I see.